modules/pages.xqm

```diff
@@ -501,6 +504,69 @@ declare function pages:generate-short-title($node, $model) as xs:string? {
     )[. ne ''][1]
 };
 
+(: Generate page breadcrumbs :)
+declare function pages:breadcrumb($node, $model){
+  pages:generate-breadcrumbs(substring-after(request:get-uri(), $app:APP_ROOT))
+};
+
+declare function pages:generate-breadcrumbs($uri as xs:string) as element(div) {
+  <div class="hsg-breadcrumb-wrapper">
+    <ol class="breadcrumb">
+      {
+        site:call-with-parameters-for-uri-steps($uri, $site:config, pages:generate-breadcrumb-item#1)
+      }
+    </ol>
+  </div>
+};
+
+declare function pages:generate-breadcrumb-item($state as map(*)) as element(li)*{
+  let $uri := $state?current-url
+  let $app-root := 
+    try {$app:APP_ROOT} 
+    catch * {
+      (: Assume APP_ROOT is '/exist/apps/hsg-shell'; Needed for xqsuite testing, 
+         since there is no context for calls to e.g. request:get-header(). :)
+      '/exist/apps/hsg-shell'
+    }
+  let $full-url := $app-root || $uri
+  let $page-template := $state?page-template
+  let $parameters as map(*)? := 
+    map:merge(
+      (
+        let $param-names as xs:string* := try {request:get-parameter-names()} catch err:XPDY0002 {()}
+        for $param-name in $param-names[. = ('region', 'subject')] (: filter necessary to avoid e.g. section-id being over-written :)
+        return map{
+          $param-name: request:get-parameter($param-name, '')
+        },
+        $state?parameters,
+        for $param in doc($page-template)//*[@data-template eq 'pages:breadcrumb']/@*[starts-with(name(.), 'data-template-')]
+        return map{
+          name($param) => substring-after('data-template-'):
+          string($param)
+        }
+      ), map{'duplicates': 'use-last'}
+    )
+  let $publication-id := $parameters?publication-id
+  let $breadcrumb-title as function(*)? := $config:PUBLICATIONS?($publication-id)?breadcrumb-title
+  let $label := 
+      if ($uri eq '/')
+        then "Home"
+      else if (doc($page-template)//*[@id eq 'breadcrumb-title'])
+        then doc($page-template)//*[@id eq 'breadcrumb-title']/node()
+      else if (exists($breadcrumb-title)) 
+        then $breadcrumb-title($parameters) 
+      else if ($config:PUBLICATIONS?($publication-id)?title)
+        then $config:PUBLICATIONS?($publication-id)?title
+      else 
+        "Office of the Historian"
+  return
+    <li>
+      <a href="{$full-url}">{" ", $label, " "
```

Comment:
No need for the extra whitespace here. All spaces can be controlled via CSS.

Reply:
I wanted to ensure that the breadcrumbs matched the layout of the old breadcrumbs exactly (to the extent that the breadcrumb links themselves were the same), and without having the edit the CSS (which would change the old as well as the new behaviour, making comparison difficult).  Plus, while I can edit CSS, it's not my strongest skill!

I'd be happy to see the back of these extraneous spaces.  Do we fix them now, or wait until we have some css features to correct?